### PR TITLE
Added jtreg-buffer testsuite

### PR DESCRIPTION
**Pull request by judovana**
based on:
https://github.com/adoptium/aqa-tests/issues/5195
https://github.com/adoptium/aqa-tests/pull/5167

### CODE REVIEW

**Comment by judovana** [comment]
s390x     https://ci.adoptium.net/view/Test_grinder/job/Grinder/9927
aarch64 https://ci.adoptium.net/view/Test_grinder/job/Grinder/9928/

**Review by karianna — APPROVED** [review]

**Review by karianna — COMMENTED** [review]
Need to understand why :-)

**Comment by judovana** [comment]
> Need to understand why :-)
Yah. Taht it honest question, and not easy one indeed!   Various reasons, from dummy ones like:   Original author quit,  original author forget, original author was lazy or simply not brave enough to push it.  And later,  no one had time to take it over.  Not exactly an good excuse, but is covering about half of the tests. I would silently nod in agreeing to anybody refusing to accept  this test set for this reason. But I tried to do quite a few in past, and burned out on it. Other reasons  are  weird corner cases, which do not run everywhere. Most of them are currenly labeled as `var.rh.jdk` although better would be `rhel.and.fedora`. Others may get this label once I see them to fail on some Adoptium combination I had not yet tried. Some have security manager. And as that is going to be dropped, but not dropped, or at least not soon, usptream is not accepting new tests with SM. Soem of them were upstreamed in form, that the  intersection of covered code was not identical, or illegible for backport. So we  decided to keep also orignal version (see speed)
Main of all is speed. upstreaming test is sometimes lengthy  (and triple it when you porting it down to the JDK where you need it most), and one need a testcase asap. That is actually why this repo was ever created.  Howerv... from there, is it super easy to slip... goto "dummy ones"; And thats why it is not empty, but actually grown a bit.

**Comment by sophia-guo** [comment]
Is this a temporary solution ( adding to functional instead of openjdk) ? What's your suggestion @smlambert ?

**Comment by smlambert** [comment]
I would prefer tests that were intended to be upstreamed to openjdk to be put into the openjdk 'group' in AQAvit rather than functional to avoid having to move them later.

File: functional/jtreg-buffer/playlist.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+               $(TEST_ROOT)$(D)functional$(D)jtreg-buffer$(D)jtreg-buffer$(D)run.sh "$(TEST_JDK_HOME)" ; $(TEST_STATUS)
+          </command>
+          <levels>
+               <level>dev</level>
+          </levels>
+          <groups>
+               <group>functional</group>
+          </groups>
+     </test>
+     <test>
```
Comment (smlambert) [inline functional/jtreg-buffer/playlist.xml:36]: Catch me up on the logic of this, I assumed that we have the 2 targets jtreg-buffer and jtreg-buffer_jtreg to allow vendors to run all tests on all platforms without the run.sh.  The main issue with the run.sh was not because readlink -f misbehaves, but that run.sh may at any point start including other behaviour that is unwanted.

Comment (judovana) [reply functional/jtreg-buffer/playlist.xml:36]: In this case it would be https://github.com/rh-openjdk/jtreg-buffer/blob/main/run.sh#L121  which would otherwise be replciated everywhere.  run.sh is from rh-openjdk organization, so it should really not do anything bad. 

(and the https://github.com/rh-openjdk/jtreg-buffer/blob/main/run.sh#L111, which is disabled in aqavit, but hsoudl be allowed to be enabled)

**Comment by smlambert** [comment]
JDK8 sparcv9_solaris: https://ci.adoptium.net/view/Test_grinder/job/Grinder/10231
JDK11 aarch64_linux: https://ci.adoptium.net/view/Test_grinder/job/Grinder/10221
JDK11 aarch64_mac: https://ci.adoptium.net/view/Test_grinder/job/Grinder/10222/ - 2 failures exit code 134, perhaps more info in jtr file?
```
11:21:26  FAILED: test/reproducers/6804996/Test.java
11:21:27  FAILED: test/reproducers/6823373/Test.java
```
JDK11 x86-64_linux: https://ci.adoptium.net/view/Test_grinder/job/Grinder/10223
JDK21 x86-64_windows: https://ci.adoptium.net/view/Test_grinder/job/Grinder/10230

**Comment by judovana** [comment]
That is the consequence of  missing `-javaoption:-Djava.security.manager=allow"` as that is the direct run without run.sh

**Comment by judovana** [comment]
> That is the consequence of missing `-javaoption:-Djava.security.manager=allow"` as that is the direct run without run.sh
```
#Test Results (version 2)
#Wed Jun 05 11:21:27 EDT 2024
#-----testdescription-----
$file=/Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer/test/reproducers/6823373/Test.java
$root=/Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer
keywords=bug6823373 shell
requires=var.msys2.enabled \=\= "false"
run=USER_SPECIFIED shell Test.sh\n
source=Test.java
title=some splash issue

#-----environment-----

#-----testresult-----
description=file\:/Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer/test/reproducers/6823373/Test.java
elapsed=339 0\:00\:00.339
end=Wed Jun 05 11\:21\:27 EDT 2024
environment=regtest
execStatus=Failed. Execution failed\: exit code 134
harnessLoaderMode=Classpath Loader
harnessVariety=Full Bundle
hostname=localhost
javatestOS=Mac OS X 14.1 (aarch64)
javatestVersion=6.0-ea+b24-2024-05-06-${BUILT_FROM_COMMIT}
jtregVersion=jtreg 7.3.1 dev 1
script=com.sun.javatest.regtest.exec.RegressionScript
sections=script_messages shell
start=Wed Jun 05 11\:21\:26 EDT 2024
test=test/reproducers/6823373/Test.java
testJDK=/Users/admin/workspace/workspace/Grinder/jdkbinary/j2sdk-image/Contents/Home/bin/..
testJDK_OS=[name\:Mac OS X,arch\:aarch64,version\:14.1,family\:mac,simple_arch\:aarch64,simple_version\:14.1,processors\:3,maxMemory\:5368709120,maxSwap\:0]
testJDK_os.arch=aarch64
testJDK_os.name=Mac OS X
testJDK_os.version=14.1
totalTime=339
user.name=admin
work=/Users/admin/workspace/workspace/Grinder/aqa-tests/TKG/output_17176008656601/jtreg-buffer_jtreg_0/work/test/reproducers/6823373

#section:script_messages
----------messages:(4/321)----------
JDK under test: /Users/admin/workspace/workspace/Grinder/jdkbinary/j2sdk-image/Contents/Home/bin/..
openjdk version "11.0.24-beta" 2024-07-16
OpenJDK Runtime Environment Temurin-11.0.24+5-202405311316 (build 11.0.24-beta+5-ea)
OpenJDK 64-Bit Server VM Temurin-11.0.24+5-202405311316 (build 11.0.24-beta+5-ea, mixed mode)

#section:shell
----------messages:(5/180)----------
command: shell Test.sh
reason: User specified action: run shell Test.sh 
started: Wed Jun 05 11:21:26 EDT 2024
finished: Wed Jun 05 11:21:27 EDT 2024
elapsed time (seconds): 0.339
----------System.out:(3/263)----------
/Users/admin/workspace/workspace/Grinder/aqa-tests/TKG/output_17176008656601/jtreg-buffer_jtreg_0/work/scratch
/Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer/test/reproducers/6823373/splash.jpg -> ./splash.jpg
run  failed
----------System.err:(24/1989)----------
openjdk version "11.0.24-beta" 2024-07-16
OpenJDK Runtime Environment Temurin-11.0.24+5-202405311316 (build 11.0.24-beta+5-ea)
OpenJDK 64-Bit Server VM Temurin-11.0.24+5-202405311316 (build 11.0.24-beta+5-ea, mixed mode)
*** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArray0 objectAtIndex:]: index 0 beyond bounds for empty array'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000182aae800 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x00000001825a5eb4 objc_exception_throw + 60
	2   CoreFoundation                      0x0000000182ad7c70 CFArrayApply + 0
	3   libsplashscreen.dylib               0x0000000100478a24 SplashNSScreen + 36
	4   libsplashscreen.dylib               0x0000000100477970 __SplashGetScaledImageName_block_invoke + 44
	5   Foundation                          0x0000000183b5e2d8 __NSThreadPerformPerform + 264
	6   CoreFoundation                      0x0000000182a39cfc __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
	7   CoreFoundation                      0x0000000182a39c90 __CFRunLoopDoSource0 + 176
	8   CoreFoundation                      0x0000000182a39a00 __CFRunLoopDoSources0 + 244
	9   CoreFoundation                      0x0000000182a385f0 __CFRunLoopRun + 828
	10  CoreFoundation                      0x0000000182a37c5c CFRunLoopRunSpecific + 608
	11  libjli.dylib                        0x0000000100559af8 CreateExecutionEnvironment + 400
	12  libjli.dylib                        0x0000000100555f54 JLI_Launch + 1196
	13  java                                0x00000001000d3b90 main + 396
	14  dyld                                0x00000001825e10e0 start + 2360
)
libc++abi: terminating due to uncaught exception of type NSException
/Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer/test/reproducers/6823373/Test.sh: line 20:  1333 Abort trap: 6           $JAVA -splash:splash.jpg -showversion Test > LOG
----------rerun:(21/1426)*----------
cd /Users/admin/workspace/workspace/Grinder/aqa-tests/TKG/output_17176008656601/jtreg-buffer_jtreg_0/work/scratch && \\
HOME=/Users/admin \\
PATH=/bin:/usr/bin:/usr/sbin \\
TESTFILE=/Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer/test/reproducers/6823373/Test.java \\
TESTSRC=/Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer/test/reproducers/6823373 \\
TESTSRCPATH=/Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer/test/reproducers/6823373 \\
TESTCLASSES=/Users/admin/workspace/workspace/Grinder/aqa-tests/TKG/output_17176008656601/jtreg-buffer_jtreg_0/work/classes/test/reproducers/6823373 \\
TESTCLASSPATH=/Users/admin/workspace/workspace/Grinder/aqa-tests/TKG/output_17176008656601/jtreg-buffer_jtreg_0/work/classes/test/reproducers/6823373 \\
COMPILEJAVA=/Users/admin/workspace/workspace/Grinder/jdkbinary/j2sdk-image/Contents/Home/bin/.. \\
TESTJAVA=/Users/admin/workspace/workspace/Grinder/jdkbinary/j2sdk-image/Contents/Home/bin/.. \\
TESTVMOPTS= \\
TESTTOOLVMOPTS= \\
TESTJAVACOPTS= \\
TESTJAVAOPTS= \\
TESTTIMEOUTFACTOR=2.0 \\
TESTROOT=/Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer \\
FS=/ \\
PS=: \\
NULL=/dev/null \\
    sh \\
        /Users/admin/workspace/workspace/Grinder/aqa-tests/functional/jtreg-buffer/jtreg-buffer/test/reproducers/6823373/Test.sh
result: Failed. Execution failed: exit code 134


test result: Failed. Execution failed: exit code 134
```

Hm Maybe not

**Comment by judovana** [comment]
1

**Comment by judovana** [comment]
It seems that all except mac is passing. I will disable those two tests on mac I think.

**Comment by judovana** [comment]
It seems that this one is not good enough for geenral testing. It seems there will  be introduced vendor-specific pipeline, so twill go there. Keeping open for now..

**Comment by judovana** [comment]
This will be moved to the vendor-specific pipeline, thus commiting elsewhere: https://github.com/rh-openjdk/jtreg-buffer/pull/23
